gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -399,9 +408,58 @@ message VehiclePosition {
   // The precision of occupancy_percentage should be low enough that you can't track a single person boarding and alighting for privacy reasons.
+  // If multi_carriage_status is populated with per-carriage occupancy_percentage, 
+  // then this field should describe the entire vehicle with all carriages accepting passengers considered.
   // This field is still experimental, and subject to change. It may be formally adopted in the future.
   optional uint32 occupancy_percentage = 10;
 
+  // Carriage specific details, used for vehicles composed of several carriages
+  message CarriageStatus {
+
+    // User visible label that may be shown to the passenger to help identify
+    // the carriage. Example: "7712", "Car ABC-32", etc...
+    optional string label = 1;
+
+    // Occupancy status for this given carriage, in this vehicle
+    optional OccupancyStatus occupancy_status = 2 [default = NO_DATA_AVAILABLE];
+
+    // Occupancy percentage for this given carriage, in this vehicle
+    // Follows the same rules as "VehiclePosition.occupancy_percentage"
+    // -1 in case data is not available (as protobuf defaults to 0 otherwise)
+    optional int32 occupancy_percentage = 3 [default = -1];
+
+    // Identifies the order of this carriage with respect to the other
+    // carriages in the vehicle's list of CarriageStatus. The values must
+    // be non-negative, and there cannot be duplicate values in the list of
+    // CarriageStatus. The values must be set such that the lowest value
+    // corresponds to the first carriage in the direction of travel, the
+    // second-lowest value corresponds to the second carriage in the direction
+    // of travel and so forth. For example, the first carriage in the direction
+    // of travel could have a carriage_sequence of 4, 
+    // the second carriage could have a carriage_sequence of 10, the third  
+    // carriage could have a carriage_sequence of 105, and so forth.
+    // Even if protobuf keeps sequencing, it will help with feed validation
+    optional uint32 carriage_sequence = 4;
+
+    // The extensions namespace allows 3rd-party developers to extend the
+    // GTFS Realtime Specification in order to add and evaluate new features and
+    // modifications to the spec.
+    extensions 1000 to 1999;
+
+    // The following extension IDs are reserved for private use by any organization.
+    extensions 9000 to 9999;
+  }
+
+  // Details of the multiple carriages of this given vehicle.
+  // The first occurrence represents the first carriage of the vehicle, 
+  // given the current direction of travel. 
+  // The number of occurrences of the multi_carriage_details 
+  // field represents the number of carriages of the vehicle.
+  // It also includes non boardable carriages, 
+  // like engines, maintenance carriages, etc… as they provide valuable 
+  // information to passengers about where to stand on a platform
+  repeated CarriageStatus multi_carriage_status = 11;
```

Comment:
Good catch. I have renamed the field as multi_carriage_details. I'd like to make sure the field name states that it should be used only in context of multiple carriages. So we prevent the case where a mono carriage vehicle relies on this field, instead of the vehicle one.